### PR TITLE
Fix BZ 64080 - graceful close

### PR DESCRIPTION
**Pull request by markt-asf**
Fixes https://bz.apache.org/bugzilla/show_bug.cgi?id=64080

Doing this via a PR to give folks a chance to review the changes first

### CODE REVIEW

File: java/org/apache/tomcat/util/net/AbstractEndpoint.java [hunk 1]
```diff
@@ -1302,6 +1320,16 @@ protected long countDownConnection() {
...
         if (bindState == BindState.BOUND_ON_START) {
+            // Stop accepting new connections
+            acceptor.stop(-1);
+            // Release locks that may be preventing the acceptor from stopping
+            releaseConnectionLatch();
+            unlockAccept();
+            // Signal to any multiplexed protocols (HTTP/2) that they may wish
+            // to stop accepting new streams
+            getHandler().pause();
+            // Update the bindSatte. This has the side-effect of disabling
```
Comment (martin-g) [inline java/org/apache/tomcat/util/net/AbstractEndpoint.java:1331]: s/bindSatte/bindState/

File: java/org/apache/tomcat/util/net/AbstractEndpoint.java [hunk 2]
```diff
@@ -1312,6 +1340,30 @@ public final void closeServerSocketGraceful() {
...
+     * Wait for the client connections to the server to close gracefully. The
+     * method will return when all of the client connections have closed or the
+     * method has been waiting for {@code waitTimeMillis}.
+     *
+     * @param waitMillis    The maximum time to wait in milliseconds for the
+     *                      client connections to close.
+     *
+     * @return The wait time, if any remaining when the method returned
+     */
+    public final long closeConnectionsAwait(long waitMillis) {
```
Comment (martin-g) [inline java/org/apache/tomcat/util/net/AbstractEndpoint.java:1353]: Maybe the method name should be `awaitConnectionsClose` ?
Now it sounds like it closes and waits, but actually it just waits.

**Review by martin-g — APPROVED** [review]

File: java/org/apache/tomcat/util/net/AbstractEndpoint.java (hunk 1 above)
Comment (markt-asf) [reply java/org/apache/tomcat/util/net/AbstractEndpoint.java]: Fixed. Tx for the review.

File: java/org/apache/tomcat/util/net/AbstractEndpoint.java (hunk 2 above)
Comment (markt-asf) [reply java/org/apache/tomcat/util/net/AbstractEndpoint.java]: Good suggestion. Fixed.

**Comment by rmaucher** [comment]
Looks ok, at least I can't see right now why it would be bad.

**Pull request closed without merging**